elixir/apps/web/lib/web/live/settings/directory_sync.ex

```diff
@@ -319,6 +319,27 @@ defmodule Web.Settings.DirectorySync do
     end
   end
 
+  def handle_event("sync_directory", %{"id" => id, "type" => "okta"}, socket) do
+    case Oban.insert(Domain.Okta.Sync.new(%{"directory_id" => id})) do
+      {:ok, _job} ->
+        {:noreply, put_flash(socket, :info, "Directory sync has been queued successfully.")}
+
+      {:error, reason} ->
+        Logger.error("Failed to enqueue Okta sync job", id: id, reason: inspect(reason))
+        {:noreply, put_flash(socket, :error, "Failed to queue directory sync.")}
+    end
+  end
+
+  def handle_event("sync_directory", %{"id" => _id, "type" => type}, socket)
+      when type in ["entra", "google"] do
+    # TODO: Implement directory sync functionality for Entra and Google
+    {:noreply, put_flash(socket, :info, "Directory sync for #{type} coming soon.")}
+  end
+
+  def handle_event("sync_directory", %{"id" => _id, "type" => _type}, socket) do
+    {:noreply, put_flash(socket, :error, "Unknown directory type.")}
+  end
+
```

Comment:
```suggestion
```

Think these cases can't be hit in practice

Reply:
> should hopefully have their own implementations

Yep they do, just right above.

> I'm curious why leaving a catch-all in would be a bad thing though?

I suppose it's not _necessarily_ a bad thing, but I don't believe it's possible to actually hit this path from the UI under any condition. I was thinking it's probably more clear if we let it crash and get a nice error notification about it rather than fail silently, because that would certainly be highly suspect.